### PR TITLE
Committed all the files to the Develop  branch

### PR DESCRIPTION
**Pull request by tavisca-rpandey**

### CODE REVIEW

File: ShoppingCart/ShoppingCart/CartItem.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿namespace ShoppingCart
+{
+    public class CartItem
+    {
+        public Product item { get; private set; }
```
Comment (tavisca-pkadam) [inline ShoppingCart/ShoppingCart/CartItem.cs:5]: naming Product instance as "item" seems a bit confusing

File: ShoppingCart/ShoppingCart/Cart.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        public List<CartItem> CartItemList{ get; private set; }
+        public double TotalCartCost { get; private set; }
+
+        public double DiscountedTotalCost { get; private set; }
+
+        private Discounts _discounts;
+
+        public Cart()
+        {
+                // Constructor for the Cart objects
```
Comment (tavisca-nrajpurohit) [inline ShoppingCart/ShoppingCart/Cart.cs]: Unnecessary Comment! (as it was obvious that this was a constructor)
Check Shared "Shared Clean Code Doc"for Commenting Rules!

File: ShoppingCart/ShoppingCart/Discounts.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections.Generic;
+
+namespace ShoppingCart
+{
+    public class Discounts
+    {
+        public Dictionary<string, double> discountTable { get; private set; }
```
Comment (tavisca-nrajpurohit) [inline ShoppingCart/ShoppingCart/Discounts.cs]: Naming Violation : Public properties/variables must be named in Pascal case.
suggestion : discountTable - > DiscountTable

**Review by tavisca-nrajpurohit — CHANGES_REQUESTED** [review]
Reviewed for naming conventions and comments.

File: ShoppingCart/ShoppingCart/CartItem.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public class CartItem
+    {
+        public Product item { get; private set; }
+        public double Quantity { get; private set; }
+        public double TotalItemCost { get; private set; }
+
+        public CartItem(Product product, double quantity)
+        {
+            item = new Product(product.Name,product.PricePerUnit);
```
Comment (tavisca-agundawar) [inline ShoppingCart/ShoppingCart/CartItem.cs]: Unnecessary creation of new object. item = product; should move the reference to the item object.

File: ShoppingCart/SoppingCartXtest/UnitTest1.cs [hunk 5]
```diff
@@ -0,0 +1,154 @@
+using System;
+using Xunit;
+using ShoppingCart;
+using System.Collections.Generic;
+
+namespace SoppingCartXtest
+{
+    public class UnitTest1
```
Comment (tavisca-pkadam) [inline ShoppingCart/SoppingCartXtest/UnitTest1.cs:8]: {ClassName}Fixture

Comment (tavisca-pkadam) [reply ShoppingCart/SoppingCartXtest/UnitTest1.cs:8]: Try Segregation of Tests For Different Classes

**Pull request merged**